src/ngraph/graph_util.cpp

```diff
@@ -912,3 +912,20 @@ bool ngraph::replace_output_update_name(Output<Node> output, const Output<Node>&
     }
     return false;
 }
+
+bool ngraph::replace_node_update_name(std::shared_ptr<Node> target,
+                                      std::shared_ptr<Node> replacement)
+{
+    for (auto& output : target->output(0).get_target_inputs())
+    {
+        if (as_type<ngraph::op::Parameter>(replacement->input_value(0).get_node()) &&
+            as_type<op::Result>(output.get_node()))
```

Comment:
No, we can replace node with another node even if output is Result. 
Just keep this:
```suggestion
    replace_node(target, replacement);
    replacement->set_friendly_name(target->get_friendly_name());
    copy_runtime_info(target, replacement);
    return true;
```

Reply:
that is only a check to avoid replacement for single layer graph (to help OV unittests).

Reply:
If you replace smth with node you won't have empty graph.

Reply:
yes, but we do not want this for single layer graph as this will fail OV unittests. (eg: param->squeeze->result becomes param->reshape->result, which fails OV unittests).